### DISCUSSION TITLE
`extends` property in `onboardingConfig` is ignored

### DISCUSSION THREAD

**Discussion by chovyy** [renovatebot/renovate#28679] [Request Help] [answered]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab

### Please tell us more about your question or problem

I'm running Renovate on a self-hosted GitLab using the [Renovate Runner](https://gitlab.com/renovate-bot/renovate-runner/).

I have the following config file:

```json
{
    "autodiscover": true,
    "onboardingConfig": {
        "$schema": "https://docs.renovatebot.com/renovate-schema.json",
        "extends": ["config:recommended", "local>mylocal/project"]
    }
}
```
But in onboarded projects the `package.json` is always missing the local reference:
```json
{
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "extends": ["config:recommended"]
}
```

I have tried other properties. Those are applied correctly. 

All Renovate runs fail without our local presets, so it is quite annoying to always start without them.

**UPDATE:** I found the relevant part in the logs. It seems that the Env config contains that part that overwrites the File config, but this was not set by me. Why are there default configuration that cannot be overwritten within the File config? The order should be: *default, file, cli, env*. and not *file, cli, env (including default)*.

### Logs (if relevant)

```
DEBUG: File config
       "config": {
         "autodiscover": true,
         "onboardingConfig": {
           "$schema": "https://docs.renovatebot.com/renovate-schema.json",
           "extends": ["config:recommended", "local>mylocal/project"]
         }
       }
DEBUG: CLI config
       "config": {}
DEBUG: Env config
       "config": {
         "hostRules": [],
         "repositoryCache": "enabled",
         "baseDir": "...",
         "logFile": "renovate-log.ndjson",
         "logFileLevel": "debug",
         "onboarding": true,
         "onboardingConfig": {
           "$schema": "https://docs.renovatebot.com/renovate-schema.json",
           "extends": ["config:recommended"]
         },
         "forkProcessing": "enabled",
         "optimizeForDisabled": true,
         "platform": "gitlab",
         "endpoint": "...",
         "token": "***********"
       }
DEBUG: Combined config
       "config": {
         "autodiscover": true,
         "onboardingConfig": {
           "$schema": "https://docs.renovatebot.com/renovate-schema.json",
           "extends": ["config:recommended"]
         },
         "hostRules": [],
         "repositoryCache": "enabled",
         "baseDir": "...",
         "logFile": "renovate-log.ndjson",
         "logFileLevel": "debug",
         "onboarding": true,
         "forkProcessing": "enabled",
         "optimizeForDisabled": true,
         "platform": "gitlab",
         "endpoint": "...",
         "token": "***********"
       }
```

**Comment by rarkins** [comment]
Please edit/rewrite your original question and remove any outdated irrelevant info

**Comment by chovyy** [comment, marked as answer]
OK, I've now understood that those "defaults" come from the Renovate Runner template. So I have to overwrite that in my pipeline.
